go/processor/floria/calls.go

```diff
@@ -16,62 +16,70 @@ import (
 	"github.com/Fantom-foundation/Tosca/go/tosca"
 )
 
-func call(interpreter tosca.Interpreter, transaction tosca.Transaction, context tosca.TransactionContext, gas tosca.Gas) (tosca.Result, error) {
+func (r runContext) Call(kind tosca.CallKind, parameters tosca.CallParameters) (tosca.CallResult, error) {
 
-	blockParameters := tosca.BlockParameters{}
-
-	transactionParameters := tosca.TransactionParameters{
-		Origin:     transaction.Sender,
-		GasPrice:   transaction.GasPrice,
-		BlobHashes: []tosca.Hash{},
+	if r.depth > CallCreateDepth {
+		return tosca.CallResult{}, nil
 	}
+	r.depth++
+	defer func() { r.depth-- }()
 
-	codeHash := context.GetCodeHash(*transaction.Recipient)
-	code := context.GetCode(*transaction.Recipient)
+	codeHash := r.GetCodeHash(parameters.Recipient)
+	code := r.GetCode(parameters.Recipient)
 
 	params := tosca.Parameters{
-		BlockParameters:       blockParameters,
-		TransactionParameters: transactionParameters,
-		//Context:               runContext, todo implement
-		Kind:      tosca.Call,
-		Static:    false,
-		Depth:     0, // todo add depth check
-		Gas:       gas,
-		Recipient: *transaction.Recipient,
-		Sender:    transaction.Sender,
-		Input:     transaction.Input,
-		Value:     transaction.Value,
-		CodeHash:  &codeHash,
-		Code:      code,
+		BlockParameters:       r.blockParams,
+		TransactionParameters: r.transactionParameters,
+		Context:               r,
+		Kind:                  tosca.Call,
```

Comment:
I guess the `Kind` should be `kind` ... but this will get unearthed when testing different call types.